crates/runtime/src/logic.rs

```diff
@@ -426,4 +484,36 @@ impl VMHostFunctions<'_> {
         self.with_logic_mut(|logic| logic.registers.set(logic.limits, register_id, data))?;
         Ok(status)
     }
+
+    /// Generates a new UUID using v4 standard.
+    ///
+    /// This function generates a new UUID by calling the host. The UUID is
+    /// dependent on randomness, which is not available inside the guest
+    /// runtime. Therefore the guest needs to request this from the host.
+    ///
+    pub fn generate_uuid(&mut self, register_id: u64) -> VMLogicResult<()> {
+        let uuid = self.borrow_logic().generate_uuid();
```

Comment:
although, there's no reason to have the `VMLogic::generate_uuid` convenience method since this specialized functionality will only be used by this specialized function

```suggestion
        let uuid = Uuid::new_v4();
        self.with_logic_mut(|logic| {
```

Reply:
Ahhhh - no; this is actually used a little more widely. I think it's best we keep this calling `generate_uuid()` as that is then consistent with the rest of the usage.

Reply:
> this is actually used a little more widely

by?

Reply:
I did, and I couldn't find any references, which is why I asked, `VMLogic` isn't to be used by anything but `calimero_runtime::run` anyway, so it can't be used by anything else